crypto/riscvcap.c

```diff
@@ -120,9 +120,7 @@ size_t riscv_vlen(void)
     return vlen;
 }
 
-# if defined(__GNUC__) && __GNUC__>=2
-__attribute__ ((constructor))
-# endif
+/* Explicitly made non-constructor due to init sequence issue */
```

Comment:
Please drop this comment.

Looking at the whole picture, I would be inclined to drop the the constructor attribute from OPENSSL_cpuid_setup() on all architectures. It should be consistent across all of them.